src/mirakuru/base.py

```diff
@@ -129,9 +130,26 @@ def __init__(
 
         self._uuid = str(uuid.uuid1())
 
+        # Python doesn't guarantee to call __del__ methods on process
+        # termination so here we catch INT, TERM and QUIT system signals
+        # and try to finally kill the subprocess if it's still running yet.
+
+        # We create a weak reference to allow garbage collection in case
+        # executor is deleted before the current process termination.
+        self_weak = weakref.ref(self)
+
+        def on_terminate(signum, frame):
+            executor = self_weak()
+            if executor:
+                executor.kill()
+
+        signal.signal(signal.SIGINT, on_terminate)
```

Comment:
Any signal handler that might have been installed by the application using mirakuru, or by a different `SimpleExecutor` instance. https://docs.python.org/2.7/library/signal.html#signal.signal returns the previous handler if it's a Python function.